### DISCUSSION TITLE
Why can't we set the id of an element?

### DISCUSSION THREAD

**Discussion by reinvanimschoot** [tailwindlabs/headlessui#851] [Help]
I'm trying to do something like this

```ts
     <Popover.Button id="__more_menu__">
        <DotsHorizontalIcon />
      </Popover.Button>
```

But the id that gets attached is `headlessui-popover-button-6`.

If I am doing something wrong, please point me in the right direction. But if this is intentional, I really do not understand this design choice.

**Comment by ortzinator** [comment]
Why do you need to set the id?

**Reply by addepar-ioc-jeff** [reply]
So we can use our own label elements

**Comment by briavicenti** [comment]
Any update on this? We're also trying to set a label for a popover, and are having a hard time linking them together properly.
